x-pack/legacy/plugins/reporting/export_types/csv_from_savedobject/server/create_job/index.ts

```diff
@@ -4,4 +4,95 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-export { createJobFactory } from './create_job';
+import { notFound, notImplemented } from 'boom';
+import { get } from 'lodash';
+import { PLUGIN_ID, CSV_FROM_SAVEDOBJECT_JOB_TYPE } from '../../../../common/constants';
+import { cryptoFactory, LevelLogger } from '../../../../server/lib';
+import { ImmediateCreateJobFn, ServerFacade, RequestFacade } from '../../../../types';
+import {
+  SavedObject,
+  SavedObjectServiceError,
+  SavedSearchObjectAttributesJSON,
+  SearchPanel,
+  TimeRangeParams,
+  VisObjectAttributesJSON,
+  JobDocPayloadPanelCsv,
+  JobParamsPanelCsv,
+} from '../../types';
+import { createJobSearch } from './create_job_search';
+
+interface VisData {
+  title: string;
+  visType: string;
+  panel: SearchPanel;
+}
+
+export const createJobFactory = function createJobFactoryFn(
+  server: ServerFacade
+): ImmediateCreateJobFn<JobParamsPanelCsv> {
+  const crypto = cryptoFactory(server);
+  const logger = LevelLogger.createForServer(server, [
+    PLUGIN_ID,
+    CSV_FROM_SAVEDOBJECT_JOB_TYPE,
+    'create-job',
+  ]);
+
+  return async function createJob(
+    jobParams: JobParamsPanelCsv,
+    headers: any,
+    req: RequestFacade
+  ): Promise<JobDocPayloadPanelCsv> {
+    const { savedObjectType, savedObjectId } = jobParams;
+    const serializedEncryptedHeaders = await crypto.encrypt(headers);
+    const client = req.getSavedObjectsClient();
+
```

Comment:
Might want to change this over to `await`'s at some point in the future since that seems to be our desired async pattern. Just stood out to me since it's Promise chains and not that.